code/__HELPERS/icons.dm

```diff
@@ -1195,87 +1181,13 @@ GLOBAL_LIST_INIT(freon_color_matrix, list("#2E5E69", "#60A2A8", "#A1AFB1", rgb(0
-	/// if successful, this looks like "icons/path/to/dmi_file.dmi"
-	var/icon_path = ""
-
-	if(isatom(icon) || istype(icon, /image) || istype(icon, /mutable_appearance))
-		var/atom/atom_icon = icon
-		icon = atom_icon.icon
-		//atom icons compiled in from 'icons/path/to/dmi_file.dmi' are weird and not really icon objects that you generate with icon().
-		//if theyre unchanged dmi's then they're stringifiable to "icons/path/to/dmi_file.dmi"
-
-	if(isicon(icon) && isfile(icon))
-		//icons compiled in from 'icons/path/to/dmi_file.dmi' at compile time are weird and arent really /icon objects,
-		///but they pass both isicon() and isfile() checks. theyre the easiest case since stringifying them gives us the path we want
-		var/icon_ref = "\ref[icon]"
-		var/locate_icon_string = "[locate(icon_ref)]"
-
-		icon_path = locate_icon_string
-
-	else if(isicon(icon) && "[icon]" == "/icon")
-		// icon objects generated from icon() at runtime are icons, but they ARENT files themselves, they represent icon files.
-		// if the files they represent are compile time dmi files in the rsc, then
-		// the rsc reference returned by fcopy_rsc() will be stringifiable to "icons/path/to/dmi_file.dmi"
-		var/rsc_ref = fcopy_rsc(icon)
-
-		var/icon_ref = "\ref[rsc_ref]"
-
-		var/icon_path_string = "[locate(icon_ref)]"
-
-		icon_path = icon_path_string
-
-	else if(istext(icon))
-		var/rsc_ref = fcopy_rsc(icon)
-		//if its the text path of an existing dmi file, the rsc reference returned by fcopy_rsc() will be stringifiable to a dmi path
 
-		var/rsc_ref_ref = "\ref[rsc_ref]"
-		var/rsc_ref_string = "[locate(rsc_ref_ref)]"
-
-		icon_path = rsc_ref_string
-
-	if(is_valid_dmi_file(icon_path))
-		return icon_path
-
-	return FALSE
-
-/**
- * generate an asset for the given icon or the icon of the given appearance for [thing], and send it to any clients in target.
- * Arguments:
- * * thing - either a /icon object, or an object that has an appearance (atom, image, mutable_appearance).
- * * target - either a reference to or a list of references to /client's or mobs with clients
- * * icon_state - string to force a particular icon_state for the icon to be used
- * * dir - dir number to force a particular direction for the icon to be used
- * * frame - what frame of the icon_state's animation for the icon being used
- * * moving - whether or not to use a moving state for the given icon
- * * sourceonly - if TRUE, only generate the asset and send back the asset url, instead of tags that display the icon to players
- * * extra_clases - string of extra css classes to use when returning the icon string
- */
-
-/proc/icon2html(atom/thing, client/target, icon_state, dir = SOUTH, frame = 1, moving = FALSE, sourceonly = FALSE)
+/proc/icon2html(thing, target, icon_state, dir = SOUTH, frame = 1, moving = FALSE, sourceonly = FALSE)
 	if (!thing)
 		return
 
 	var/key
-	var/icon/icon2collapse = thing
+	var/icon/I = thing
```

Comment:
1 letter var

Reply:
Uhhh I never touched this, I will revert changes on this file